src/overlay.jsx

```diff
@@ -39,36 +40,35 @@ const Overlay = React.createClass({
   componentWillReceiveProps (nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
     this.setState({muiTheme: newMuiTheme});
+    if (this.state.show !== nextProps.show) {
+      this.setState({show:nextProps.show}, this._applyAutoLockScrolling);
+    }
   },
 
   propTypes: {
     autoLockScrolling: React.PropTypes.bool,
-    show: React.PropTypes.bool,
-    transitionEnabled: React.PropTypes.bool,
+    show: React.PropTypes.bool.isRequired,
     style: React.PropTypes.object,
+    transitionEnabled: React.PropTypes.bool,
   },
 
   getDefaultProps() {
     return {
       autoLockScrolling: true,
       transitionEnabled: true,
+      style:{},
     };
   },
+  componentWillMount() {
+    if (this.state.show !== this.props.show) {
+      this.setState({show:this.props.show}, this._applyAutoLockScrolling);
```

Comment:
The setState seems useless here. I guess the best thing to do is to just call
`_applyAutoLockScrolling` on the `componentDidMount()`